### ISSUE TITLE
Automatically determine whether AssemblyInfo.cs files should be written

### ISSUE THREAD

**Issue by logiclrd** [rfennell/AzurePipelines#1374]
### Azure DevOps Extensions

Built Utilities

### Is your feature request related to a problem? Please describe.

I upgraded projects to SDK format, which by default emits AssemblyInfo.cs automatically as part of the build, and the builds started failing. The reason is that, unbeknownst to me, the build processes set up by the team use your extension to emit AssemblyInfo.cs files with the build version, and that creates a conflict with the generated ones. The thing is, it seems to me that it should be possible to _infer_ whether a project should have AssemblyInfo.cs files generated: if the .csproj is an SDK-style file _and_ it does not have `<GenerateAssemblyInfo>false</GenerateAssemblyInfo>`, then generating _another_ AssemblyInfo.cs file _will_ cause the build to fail. There are also properties that control specific attributes, so e.g. if the project has `<GenerateAssemblyVersionAttribute>false</GenerateAssemblyVersionAttribute>`, then adding an AssemblyInfo.cs file with an `AssemblyVersion` attribute won't break the build, but if that file also include `AssemblyFileVersion` it'll break, that sort of thing.

### Describe the solution you'd like

I'd like the build task for generating AssemblyInfo.cs files to version assemblies to restrain itself automatically if the project file is configured in a way that will cause conflicts.

**Comment by rfennell** [comment]
Could you not swap to using the `VersionDotNetCoreAssembliesTask`, this applies a version to a .NET Core assembly via the .csproj files based on the build number. Based on Microsoft sample from https://msdn.microsoft.com/Library/vs/alm/Build/overview

**Comment by logiclrd** [comment]
The problem is, some projects in the solution need AssemblyInfo.cs-based versioning, and some _cannot_ have AssemblyInfo.cs. I'm coming to an existing solution that is having problems now, though, so maybe there's a way to make this work that I'm unaware of. :-)

**Comment by rfennell** [comment]
Yes, I can see where you are coming from, but I am a bit confused as my `VersionAssemblies` task does not create an `AssemblyInfo.cs` file, it just updates one or more of the version fields in any pre-existing `AssemblyInfo.cs` files it finds in the specified path.

So, if you want your `assemblyinfo.cs` to be emitted by the build process, just delete the `assemblyinfo.cs`file in source control and my task will do nothing as it is run before the build tasks i.e. before the generated file exists

Also, a solution as you propose, where there is a test of the `<GenerateAssemblyInfo>false|true</GenerateAssemblyInfo>`  would mean the task needed to make assumptions. This is because at present my task know nothing of `csproj` files, it would have to try to find the correct one by walking up folder path. It should work, until someone puts two `csproj` files in the same folder for some strange reason.

This feels like too much of an edge case to be added to the task

**Comment by logiclrd** [comment]
Mm, okay. So the solution to my problem could be to rename the files that are still providing _some_ attributes.
